XVSEC and XDMA driver update for Linux 5.x

paulmnt opened the pull request:
* Remove unused ret2 argument from call to ki_complete() on Linux Kernel version >= 5.16
* Guard `get_fs()` / `set_fs` calls based on Linux Kernel version < 5.10
* Use `kernel_read()` / `kernel_write()` for Linux Kernel version >= 5.10
* Add .gitignore
* Minor fix to Makefile

paulmnt committed:
Add .gitignore

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

paulmnt committed:
Fix XVSEC/linux-kernel/drv/Makefile

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

paulmnt committed:
Fix warning on exported symbols for xvsec_drv.c

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

paulmnt committed:
Fix xvsec_util for Linux Kernel version >= 5.10

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

paulmnt committed:
Fix call to ki_complete() for Linux kernel v5.16

Unused ret2 argument has been dropped from function signature

paulmnt committed:
Update libxdma for kernel v 5.18

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

hmaarrfk commented:
Xilinx likely doesn't won't respond and might delete this post.

If you want to help get the compilation infrastructure and help on my fork, i will gladly review. An example is show to test qdma for different kernel versions

https://github.com/hmaarrfk/dma_ip_drivers/

cyyself commented:
I have tested your xvsec patch on Linux v6.3 with firesim on Xilinx VCU118 Board, but I have some issues. 

Although it was compiled successfully, the system stops responding when I use `firesim infrasetup`, which uses `xvsecctl` to replace the bitstream. If I restart the system (without power cycle FPGA) this time, it stops at BIOS. But when I switch to Kernel v5.4, it works normally.

paulmnt committed:
Update xdma for kernel v 6.3.0

vm_flags can only be modified with helper functions

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

paulmnt committed:
Update xdma for kernel v 6.4.0

 - iovec pointer iov changed to internal variable
 - device class create takes does not take pointer to module owner.

Signed-off-by: Paolo Mantovani <pmantovani@google.com>

devegali commented:
All changes are now part of master repo
